third_party/maya/lib/usdMaya/xformStack.h

```diff
@@ -0,0 +1,315 @@
+{
+public:
+    typedef PxrUsdMayaXformOpClassification OpClass;
+    typedef std::vector<OpClass> OpClassList;
+    typedef std::pair<const OpClass, const OpClass> OpClassPair;
+
+    // Internally, we use indices, because position sometimes matters...
+    // should be safe, since _ops is const.
+    static constexpr size_t NO_INDEX = std::numeric_limits<size_t>::max();
+    typedef std::pair<size_t, size_t> IndexPair;
+    typedef std::unordered_map<TfToken, IndexPair, TfToken::HashFunctor>
+            TokenIndexPairMap;
+    typedef std::unordered_map<size_t, size_t> IndexMap;
+
+
+    // Templated because we want it to work with both MEulerRotation::RotationOrder
+    // and MTransformationMatrix::RotationOrder
+    template<typename RotationOrder>
+    static RotationOrder RotateOrderFromOpType(
+            UsdGeomXformOp::Type opType,
+            RotationOrder defaultRotOrder=RotationOrder::kXYZ)
+    {
+        switch(opType) {
+            case UsdGeomXformOp::TypeRotateXYZ:
+                return  RotationOrder::kXYZ;
+            break;
+            case UsdGeomXformOp::TypeRotateXZY:
+                return  RotationOrder::kXZY;
+            break;
+            case UsdGeomXformOp::TypeRotateYXZ:
+                return  RotationOrder::kYXZ;
+            break;
+            case UsdGeomXformOp::TypeRotateYZX:
+                return  RotationOrder::kYZX;
+            break;
+            case UsdGeomXformOp::TypeRotateZXY:
+                return  RotationOrder::kZXY;
+            break;
+            case UsdGeomXformOp::TypeRotateZYX:
+                return  RotationOrder::kZYX;
+            break;
+            default:
+                return defaultRotOrder;
+            break;
+        }
+    }
+
+    PxrUsdMayaXformStack(
+            const OpClassList& ops,
+            const std::vector<std::pair<size_t, size_t> >& inversionTwins,
+            bool nameMatters=true);
+
+    // Don't want to accidentally make a copy, since the only instances are supposed
+    // to be static!
+    explicit PxrUsdMayaXformStack(const PxrUsdMayaXformStack& other) = default;
+    explicit PxrUsdMayaXformStack(PxrUsdMayaXformStack&& other) = default;
+
+    PXRUSDMAYA_API
+    OpClassList const & GetOps() const;
+
+    PXRUSDMAYA_API
+    std::vector<std::pair<size_t, size_t> > const & GetInversionTwins() const;
+
+    PXRUSDMAYA_API
+    bool GetNameMatters() const;
+
+    PXRUSDMAYA_API
+    PxrUsdMayaXformOpClassification const & operator[] (const size_t index) const;
+
+    PXRUSDMAYA_API
+    size_t GetSize() const;
+
+
+    /// \brief  Finds the index of the Op Classification with the given name in this stack
+    /// \param  opName the name of the operator classification we  wish to find
+    /// \param  isInvertedTwin the returned op classification object must match
+    ///         this param for it's IsInvertedTwin() - if an op is found that matches
+    ///         the name, but has the wrong invertedTwin status, nullptr is returned
```

Comment:
Looks like this should be "NO_INDEX is returned" instead of "nullptr is returned"